graphql/resolvers/workspaceResolver.js

```diff
@@ -118,11 +118,11 @@ module.exports = {
      * @param {{ loaders: { articles } }} context
      * @returns {Promise<*>}
      */
-    async articles (workspace, _args, context) {
-      const articles = (await Promise.all(workspace.articles.map((articleId) => context.loaders.articles.load(articleId))))
-        .filter((a) => a !== undefined) // remove deleted articles
-      articles.sort((a, b) => a.createdAt > b.createdAt ? -1 : 1)
-      return Article.complete(articles, context.loaders)
+    async articles(workspace, _args, context) {
+      return Article.getArticles({
+        filter: { _id: { $in: workspace.articles } },
+        loaders: context.loaders,
+      })
```

Comment:
C'est malin je n'y avais pas pensé 👍🏻 
Est-ce qu'il y a une différence notable dans les performances ? Si tu associes plusieurs centaines/milliers d'articles à un espace de travail ?

Reply:
Je ne sais pas, c'est encore mystérieux le fonctionnement des loaders (ça appelle `.complete()` quand même, donc ça ne change pas grand chose à la précédente manière de faire ?)

Reply:
De prime abord, je dirais que ta solution est plus rapide.
Comme le code est plus simple et que je pense qu'on a assez peu d'espace de travail avec plus de 50 articles on s'en fiche un peu. C'était plus par curiosité 😅

Reply:
Pour les data loaders ça permet de limiter l'effet des requêtes N+1

https://www.apollographql.com/tutorials/dataloaders-dgs/03-data-loaders-under-the-hood